backend/apps/progress/services/leaderboard_service.py

```
@@ -0,0 +1,155 @@
+
+logger = logging.getLogger(__name__)
+
+
+def get_redis_client():
+    redis_url = os.getenv("REDIS_URL", "redis://127.0.0.1:6379/1")
+    try:
+        # Check if we can connect
+        client = redis.from_url(redis_url, decode_responses=True, socket_connect_timeout=1)
+        client.ping()
+        return client
+    except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)